Code/Framework/AzToolsFramework/AzToolsFramework/AssetEditor/AssetEditorWidget.cpp

```diff
@@ -306,306 +218,91 @@ namespace AzToolsFramework
-                filter.append(assetTypeName);
-                filter.append(" (");
-                for (size_t i = 0, n = assetTypeExtensions.size(); i < n; ++i)
+                // Check that all tabs are clean.
+                for (int tabIndex = 0; tabIndex < m_tabs->count(); tabIndex++)
                 {
-                    const char* ext = assetTypeExtensions[i].c_str();
-                    if (ext[0] == '.')
+                    AssetEditorTab* tab = qobject_cast<AssetEditorTab*>(m_tabs->widget(tabIndex));
+                    if (tab->IsDirty() && !tab->UserRefusedSave())
                     {
-                        ++ext;
-                    }
-
-                    filter.append("*.");
-                    filter.append(ext);
-                    if (i < n - 1)
-                    {
-                        filter.append(" ");
+                        return;
                     }
                 }
-                filter.append(")");
-            }
 
-            const QString saveAs = AzQtComponents::FileDialog::GetSaveFileName(
-                AzToolsFramework::GetActiveWindow(), tr("Save As..."), m_userSettings->m_lastSavePath.c_str(), filter);
+                // Close them.
+                for (int tabIndex = m_tabs->count() - 1; tabIndex >= 0; tabIndex--)
+                {
+                    AssetEditorTab* tab = qobject_cast<AssetEditorTab*>(m_tabs->widget(tabIndex));
+                    m_tabs->removeTab(tabIndex);
+                    delete tab;
+                }
 
-            return SaveImpl(asset, saveAs);
-        }
+                CloseOnNextTick();
+            };
 
-        bool AssetEditorWidget::SaveAssetToPath(AZStd::string_view assetPath)
-        {
-            return SaveImpl(m_inMemoryAsset, assetPath.data());
-        }
+            // Ensure the UserRefusedSave flag is reset so that if this gets called twice, dirty tabs aren't ignored.
+            for (int tabIndex = m_tabs->count() - 1; tabIndex >= 0; tabIndex--)
+            {
+                AssetEditorTab* tab = qobject_cast<AssetEditorTab*>(m_tabs->widget(tabIndex));
+                tab->ClearUserRefusedSaveFlag();
+            }
 
-        bool AssetEditorWidget::SaveImpl(const AZ::Data::Asset<AZ::Data::AssetData>& asset, const QString& saveAsPath)
-        {
-            if (!saveAsPath.isEmpty())
+            if (!m_tabs->count())
             {
-                AZStd::string targetFilePath(saveAsPath.toUtf8().constData());
+                parentWidget()->parentWidget()->close();
+                return true;
+            }
 
-                if (!m_useDPE)
+            for (int tabIndex = m_tabs->count() - 1; tabIndex >= 0; tabIndex--)
+            {
+                AssetEditorTab* tab = qobject_cast<AssetEditorTab*>(m_tabs->widget(tabIndex));
+                if (!tab)
                 {
-                    m_propertyEditor->ForceQueuedInvalidation();
+                    continue;
```

Comment:
Same comment as above here

Reply:
The continue skips to the next iteration of the for loop, so no else is necessary.